transport.go

```diff
@@ -87,28 +84,39 @@ func ensureSameProtocol(hostPorts []string) (string, error) {
 }
 
 func loadTransportHostPorts(opts TransportOptions) (TransportOptions, error) {
-	if len(opts.HostPorts) == 0 && opts.HostPortFile == "" {
-		return opts, errPeerRequired
-	}
-
-	hostPorts := opts.HostPorts
+	peers := opts.HostPorts
 	if opts.HostPortFile != "" {
-		if len(hostPorts) > 0 {
+		if len(peers) > 0 {
 			return opts, errPeerOptions
 		}
-		var err error
-		hostPorts, err = parseHostFile(opts.HostPortFile)
+
+		u, err := url.Parse(opts.HostPortFile)
 		if err != nil {
-			return opts, fmt.Errorf("failed to parse host file: %v", err)
+			return opts, fmt.Errorf("could not parse peer provider URL: %v", err)
 		}
 
-		if len(hostPorts) == 0 {
-			return opts, errPeerRequired
+		ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+		defer cancel()
+
+		// Paths are a strict subset of the URL file protocol scheme. We
+		// explicate the file protocol as a key into the peer provider
+		// registry.
+		if u.Scheme == "" {
```

Comment:
Could we register the file provider under the `""` scheme and have this work without this `if`?